files/default/ec2_dev_2_volid.py

```diff
@@ -39,10 +39,18 @@ def main():
         dev = "/dev/" + dev
 
     # Get instance ID
-    instance_id = requests.get("http://169.254.169.254/latest/meta-data/instance-id").text
+    token = requests.put(
+        "http://169.254.169.254/latest/api/token",
+        headers={"X-aws-ec2-metadata-token-ttl-seconds": "300"}
+    )
+
+    headers = {}
+    if token.status_code == requests.codes.ok:
+        headers["X-aws-ec2-metadata-token"] = token.content
+    instance_id = requests.get("http://169.254.169.254/latest/meta-data/instance-id", headers=headers).text
```

Comment:
Same comment as above. Can we reuse the same function definition in a shared python file?
In any case, we have a bunch of code meant to a single specific functionality that's described in the comment `# Get instance ID`.  In these cases you should have better speaking code, here a function named `get_instance_id()`, that would let you to remove both all those lines of code and the comment as well..

Reply:
Also add a function get_imdsv2_token() in this PR: #940 to warp the part of getting token to make it more clear as explain above.
I am not reusing the same function definition in a shared python file, the reason is that this file is under files/default directory in our code. The cookbook files will be specified to put into different paths. For example, this file will be put in path '/sbin/ec2_dev_2_volid.py'. Other files will be specified to put in other places. These python files are called separately. We may not want to import another python files when we called them.